package.json

```diff
@@ -5,6 +5,20 @@
     "main": "dist/cjs/index.js",
     "module": "dist/mjs/index.js",
     "typings": "dist/mjs/index.d.ts",
+    "exports": {
+      ".": {
+        "import": "./dist/mjs/index.js",
+        "require": "./dist/cjs/index.js"
+      },
+      "./oas30": {
+        "import": "./dist/mjs/oas30.js",
+        "require": "./dist/cjs/oas30.js"
+      },
+      "./oas31": {
+        "import": "./dist/mjs/oas31.js",
+        "require": "./dist/cjs/oas31.js"
+      }
```

Comment:
Not sure why, but such imports are not appreciated by Typescript, however they are operational if ignored. ¯\_(ツ)_/¯ 

![image](https://user-images.githubusercontent.com/13189514/232337329-d7025c45-c006-4e6c-9788-66296857e108.png)

Reply:
No, actually my build fails to make DTS with such imports

```
src/open-api.ts(8,8): error TS2307: Cannot find module 'openapi3-ts/oas30' or its corresponding type declarations.
Error: error occured in dts build
DTS Build error
RollupError: Failed to compile. Check the logs above.
```

Reply:
Something is not right with it.

Reply:
Ah, I guess your tsconfig.json moduleResolution: "node" which cannot resolve the exports field. As one option, node16 would work tho.

Reply:
It is.
Could you tell me please, @koooge , does it mean, that the minimum Node version of my package (that depends on `openapi3-ts`) should be upgraded from 14 to 16 ?

Reply:
I would say node version does not matter. It is due to the config of typescript. If you use tsc with `"moduleResolution": "node"` ,  this approach would not work.

Reply:
Ok. Thank you.
But there is another problem.
Still, my ESM setup does not work with the following error regarding `./oas30` import:

```
SyntaxError: Named export 'isReferenceObject' not found. The requested module 'openapi3-ts/oas30' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'openapi3-ts/oas30';
```

I use `ts-node-esm` as a runner.

Reply:
It thinks that `/oas30` is CommonJS, @koooge .

It can be fixed by placing `package.json` file inside `./dist/mjs` having `{"type": "module"}`.

However, it does not end on it. It can not find files being imported by `oas30.js` withing having `.js`extensions in the import statements. I believe the ESM build should actually have the explicitly specified extensions. Probably the ESM compiler need to be configured. CC @pjmolina

Reply:
That's because moduleResolution node(=node10) checks "main" in pckage.json. Use `"moduleResolution": "node16"`  or bundler as a workaround. For support tsc, we need to export only "." and create dist/oas30 & dist/oas31 I suppose.

Reply:
Here is one more notice about this

```
Error: tests/unit/index.spec.ts(9,31): error TS2834: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.
```